### PR TITLE
Refactor FXIOS-10647 Remove crash alert

### PR DESCRIPTION
**Pull request by OrlaM**
## :scroll: Tickets
[Jira ticket](https://mozilla-hub.atlassian.net/browse/FXIOS-TODO)
[Github issue](https://github.com/mozilla-mobile/firefox-ios/issues/TODO)

## :bulb: Description
Remove the restore tabs alert

## :pencil: Checklist
You have to check all boxes before merging
- [ ] Filled in the above information (tickets numbers and description of your work)
- [ ] Updated the PR name to follow our [PR naming guidelines](https://github.com/mozilla-mobile/firefox-ios/wiki/Pull-Request-Naming-Guide)
- [ ] Wrote unit tests and/or ensured the tests suite is passing
- [ ] When working on UI, I checked and implemented accessibility (minimum Dynamic Text and VoiceOver)
- [ ] If needed, I updated documentation / comments for complex code and public methods
- [ ] If needed, added a backport comment (example `@Mergifyio backport release/v120`)

### CODE REVIEW

**Comment by OrlaM** [comment]
@MergifyIO backport release/v133

File: firefox-ios/Client/Frontend/Browser/BrowserViewController/Views/BrowserViewController.swift [hunk 1]
```diff
@@ -1288,40 +1277,9 @@ class BrowserViewController: UIViewController,
...
-                self.openUrlAfterRestore()
-                AppEventQueue.signal(event: .tabRestoration(self.tabManager.windowUUID))
-            }
-        )
-        self.present(alert, animated: true, completion: nil)
-        isCrashAlertShowing = true
-    }
+//    fileprivate func crashedLastLaunch() -> Bool {
+//        return logger.crashedLastLaunch
+//    }
```
Comment (MattLichtenstein) [inline firefox-ios/Client/Frontend/Browser/BrowserViewController/Views/BrowserViewController.swift:1282]: Do we need this commented out code?

**Review by MattLichtenstein — APPROVED** [review]
One question - otherwise looks good 👍

File: firefox-ios/Client/Frontend/Browser/BrowserViewController/Views/BrowserViewController.swift (hunk 1 above)
Comment (OrlaM) [reply firefox-ios/Client/Frontend/Browser/BrowserViewController/Views/BrowserViewController.swift:1282]: Nope, should be deleted 🙈

File: firefox-ios/Client/Frontend/Browser/BrowserViewController/Views/BrowserViewController.swift [hunk 2]
```diff
@@ -1288,41 +1277,6 @@ class BrowserViewController: UIViewController,
...
-
-    fileprivate func showRestoreTabsAlert() {
-        let alert = UIAlertController.restoreTabsAlert(
-            okayCallback: { _ in
-                let extra = [TelemetryWrapper.EventExtraKey.isRestoreTabsStarted.rawValue: true]
-                TelemetryWrapper.recordEvent(category: .action,
-                                             method: .tap,
-                                             object: .restoreTabsAlert,
-                                             extras: extra)
-                self.isCrashAlertShowing = false
```
Comment (lmarceau) [inline firefox-ios/Client/Frontend/Browser/BrowserViewController/Views/BrowserViewController.swift:1307]: `isCrashAlertShowing` should be removed, and it's used in one other place it seems (under `switchToTabForURLOrOpen`) 🤔

File: firefox-ios/Client/Frontend/Browser/BrowserViewController/Views/BrowserViewController.swift [hunk 3]
```diff
@@ -1288,41 +1277,6 @@ class BrowserViewController: UIViewController,
...
-    var displayedRestoreTabsAlert = false
-
-    fileprivate func crashedLastLaunch() -> Bool {
-        return logger.crashedLastLaunch
-    }
-
-    fileprivate func showRestoreTabsAlert() {
-        let alert = UIAlertController.restoreTabsAlert(
-            okayCallback: { _ in
-                let extra = [TelemetryWrapper.EventExtraKey.isRestoreTabsStarted.rawValue: true]
```
Comment (lmarceau) [inline firefox-ios/Client/Frontend/Browser/BrowserViewController/Views/BrowserViewController.swift:1302]: We should remove `TelemetryWrapper.EventExtraKey.isRestoreTabsStarted`

File: firefox-ios/Client/Frontend/Browser/BrowserViewController/Views/BrowserViewController.swift [hunk 4]
```diff
@@ -1288,41 +1277,6 @@ class BrowserViewController: UIViewController,
...
-                                             object: .restoreTabsAlert,
-                                             extras: extra)
-                self.isCrashAlertShowing = false
-                self.tabManager.restoreTabs(true)
-            },
-            noCallback: { _ in
-                let extra = [TelemetryWrapper.EventExtraKey.isRestoreTabsStarted.rawValue: false]
-                TelemetryWrapper.recordEvent(category: .action,
-                                             method: .tap,
-                                             object: .restoreTabsAlert,
```
Comment (lmarceau) [inline firefox-ios/Client/Frontend/Browser/BrowserViewController/Views/BrowserViewController.swift:1314]: `.restoreTabsAlert` Glean event should be removed as well

**Review by lmarceau — COMMENTED** [review]
@OrlaM There's some minor clean up following this PR I added as comments, let me know I can open a quick PR to do them if you wish

**Pull request merged**